### DISCUSSION TITLE
Kitty has much better performance when built from source

### DISCUSSION THREAD

**Discussion by ghost** [kovidgoyal/kitty#6505] [Q&A]
Obviously not a complaint, it's just something that struck me recently. Kitty already has stellar performance and when I compile from source it seems to be even better. Tested on an M1 mac. In Neovim, when I have multiple split buffers, the fps goes down significantly when installing through homebrew or just using one of the release binaries. Tweaking the repaint_delay setting seems to only have a minimal effect. Building from source however, scrolling is buttery smooth. I was wondering if someone could shed some light on why this might be happening or give me some pointers to where I could go looking for an answer.

**Comment by kovidgoyal** [comment]
I dont see how that's possible, the only thing I can think of
is when building from source kitty is built with all the CPU specific optimizations
for your specific CPU. But I highly doubt that can make actual user
visible difference.

**Reply by ghost** [reply]
Is there any way I could confirm? I'm really curious what could be causing this.

**Comment by madhur** [comment]
May be while building the source, you got this fix included https://github.com/kovidgoyal/kitty/issues/6479

This recent fix has improved the performance on kitty on Mac M1

**Reply by ghost** [reply]
That commit is included in the the nightly release if I'm not mistaken. Installing the nightly version does not include the performance boost that I can notice when building from source.

**Comment by kovidgoyal** [comment]
Nightly definitely has the commit as does the latest release.

**Reply by ghost** [reply]
I tested with both, but building from source still provides the smoothest scrolling experience.

**Reply by kovidgoyal** [reply]
well like i said i cant really see anything that could cause it. you can try diffing the output of env inside the source and binary versions to see if anything stands out.
